flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkChangelogModeInferenceProgram.scala

```diff
@@ -310,18 +310,26 @@ class FlinkChangelogModeInferenceProgram extends FlinkOptimizeProgram[StreamOpti
           .newBuilder()
           .addContainedKind(ModifyKind.INSERT)
 
-        // All aggregates are computed over the same window and order by is supported for only 1 field
-        val orderKeyIndex =
-          over.logicWindow.groups.get(0).orderKeys.getFieldCollations.get(0).getFieldIndex
-        val orderKeyType = over.logicWindow.getRowType.getFieldList.get(orderKeyIndex).getType
         if (
-          !FlinkTypeFactory.isRowtimeIndicatorType(orderKeyType)
-          && !FlinkTypeFactory.isProctimeIndicatorType(orderKeyType)
+          !over.logicWindow.groups.isEmpty && !over.logicWindow.groups
+            .get(0)
+            .orderKeys
+            .getFieldCollations
+            .isEmpty
```

Comment:
out of curiosity: is there a case when it is empty and `over.logicWindow.groups` is not empty?

Reply:
No that can't happen since order by is specified inside the over clause

Reply:
then I don't understand why do we need both `!groups.isEmpty && !groups.get(0).orderKeys.getFieldCollations.isEmpty` in condition?
isn't just one of them enough?

Reply:
Ideally this one condition should be enough i.e. `!groups.get(0).orderKeys.getFieldCollations.isEmpty` but the condition `!groups.isEmpty` is just a safety check. We can remove the safety check since there will always be 1 group.